### PR TITLE
Create generator support for conditional logic

### LINKED ISSUE #287: Create Condition model, serializer, generator on backend

### PR DESCRIPTION
**Pull request by Kapin**
Closes #287

### CODE REVIEW

File: src-django/api/generator.py [hunk 1]
```diff
@@ -115,12 +115,87 @@ def generate(self, parent):
...
+        self.condition_node = condition_node
+
+        if self.condition_node.node_type not in ConditionNode.CRITERIA_TYPES:
+            self.__raise_error('node type must be either EQUALS, LESS, or GREATER')
+
+    def __raise_error(self, error_string):
+        page = self.condition_node.show_if.page
+        raise ValueError('Error on page %d with display_index %d: %s' % (page.id, page.display_index, error_string))
+
+    def get_properties(self):
```
Comment (Ommy) [inline src-django/api/generator.py]: Any reason why this method isn't prefixed with double underscores?

**Comment by Ommy** [comment]
+1

**Comment by Trinovantes** [comment]
+1

**Comment by connorcimowsky** [comment]
:+1:

**Pull request merged**